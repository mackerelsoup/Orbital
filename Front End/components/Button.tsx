import { Pressable, StyleSheet, Text, View } from 'react-native';

type Props = {
    label: string;
    onPress?: () => void;
    children?: React.ReactElement
};

export default function Button({ label, onPress, children }: Props) {

    return (
        <View style={[
            styles.buttonContainer
        ]}>
            <Pressable
                style={({ pressed }) => [
                    styles.button,
                    { opacity: pressed ? 0.5 : 1 },
                ]}
                onTouchEnd={onPress}
                >
                <View style = {{flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                    <Text style={styles.buttonLabel}>{label}</Text>
                    {children}
                </View>

            </Pressable>
        </View>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        alignItems: 'center',
        justifyContent: 'center',
        margin: 8,

    },
    button: {
        width: "100%",
        alignItems: 'center',
        justifyContent: 'center',
        //borderRadius: 50,
        backgroundColor: "black"

    },
    buttonLabel: {
        color: '#fff',
        fontSize: 16,
        textAlign: 'center'
    },
});
