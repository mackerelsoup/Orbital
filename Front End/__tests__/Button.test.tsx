import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '../components/Button';
import { Text } from 'react-native';

describe('Button', () => {
  it('renders the label correctly', () => {
    const { getByText } = render(<Button label="Click me" />);
    expect(getByText('Click me')).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<Button label="Press me" onPress={onPressMock} />);
    fireEvent.press(getByText('Press me'));
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('renders children correctly', () => {
    const { getByText } = render(
      <Button label="With children">
        <Text>Child content</Text>
      </Button>
    );
    expect(getByText('With children')).toBeTruthy();
    expect(getByText('Child content')).toBeTruthy();
  });
});
