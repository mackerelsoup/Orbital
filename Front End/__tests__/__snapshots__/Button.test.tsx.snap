// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button Component matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 8,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTouchEnd={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "opacity": 1,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Snapshot Test
      </Text>
    </View>
  </View>
</View>
`;
